plugins/filter/json_patch.py

```diff
@@ -0,0 +1,176 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) Stanislav Meduna (@numo68)
+# GNU General Public License v3.0+ (see LICENSES/GPL-3.0-or-later.txt or https://www.gnu.org/licenses/gpl-3.0.txt)
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from __future__ import absolute_import, annotations, division, print_function
+from json import loads
+from typing import TYPE_CHECKING
+from ansible.errors import AnsibleFilterError
+from ansible.module_utils.common.text.converters import to_native
+
+__metaclass__ = type  # pylint: disable=C0103
+
+if TYPE_CHECKING:
+    from typing import Any, Callable, Union
+
+try:
+    import jsonpatch
+
+except ImportError as exc:
+    HAS_LIB = False
+    JSONPATCH_IMPORT_ERROR = exc
+else:
+    HAS_LIB = True
+    JSONPATCH_IMPORT_ERROR = None
+
+OPERATIONS_AVAILABLE = ["add", "copy", "move", "remove", "replace", "test"]
+OPERATIONS_NEEDING_FROM = ["copy", "move"]
+OPERATIONS_NEEDING_VALUE = ["add", "replace", "test"]
+
+
+class FilterModule:
+    """Filter plugin."""
+
+    def check_json_object(
+        self, filter_name: str, object_name: str, inp: Union[str, list, dict]
+    ):
+        if not isinstance(inp, (str, list, dict)):
```

Comment:
I would probably also accept `binary` here. `loads` also handles binary strings, and if the filter is chained with something that outputs a binary string (like the `ansible.builtin.b64decode` filter), this would make life a lot simpler for the user.

Reply:
Hmm, according to the documentation and also a quick test the `b64decode` outputs a string, not `bytes` or `bytearray`. If there is any way to pass a binary input to the filter, I am not aware of it - is there?

Reply:
There is, but only from inside another Jinja plugin and while using the 'native' engine. From Ansible itself you cannot pass byte strings, only unicode strings.

Reply:
okay in that case I'll implement accepting bytes and bytearray in the python and test it in the unit test, but no further comment on it in docs nor errors returned.